### PR TITLE
Fix tests related to join and subtensors tests

### PR DESCRIPTION
**Pull request by nouiz**

### CODE REVIEW

File: theano/tensor/tests/test_basic.py [hunk 1]
```diff
@@ -3750,7 +3752,10 @@ def get_mat(s1, s2):
             self.assertRaises(ValueError, f, get_mat(3, 4), get_mat(3, 4),
                               get_mat(2, 5))
```
Comment (lamblin) [inline theano/tensor/tests/test_basic.py]: Remove the `get_mat(...)` arguments here, since `f` does not take arguments anymore.

Comment (nouiz) [inline theano/tensor/tests/test_basic.py]: done.

**Comment by lamblin** [comment]
Well, now, `AssertionError: ValueError not raised` in FAST_COMPILE.
I think that the series of `set_value` should be done before the `if`.

**Comment by nouiz** [comment]
done

**Pull request merged**